packages/appsync-modelgen-plugin/src/types/sync.ts

```diff
@@ -0,0 +1,46 @@
+import { Types, PluginFunction as PluginFunctionAsync, CodegenPlugin as CodegenPluginAsync } from "@graphql-codegen/plugin-helpers";
+
+type PluginMapContainer = Pick<Types.GenerateOptions, 'pluginMap'>;
+type CacheContainer = Pick<Types.GenerateOptions, 'cache'>;
+
+export type SyncPluginMap<Obj extends PluginMapContainer> = Omit<Obj, 'pluginMap'> & {
+    pluginMap: {
+        [name: string]: Omit<Obj['pluginMap'][string], 'plugin'> & {
+        plugin: (
+            ...args: Parameters<Obj['pluginMap'][string]['plugin']>
+        ) => Awaited<ReturnType<Obj['pluginMap'][string]['plugin']>>;
+        };
+    };
+};
+
+export type SyncCache<Obj extends CacheContainer> = Omit<Obj, 'cache'> & {
+    cache?: (<T>(namespace: string, key: string, factory: () => T) => T) | undefined
+};
```

Comment:
These two helper types do most of the lifting here. They unwrap the Promise/Await type expectations so that the types can be used for sync versions of the target behavior.

Awaited behavior is used/allowed for:
- Plugin execution
- Profilers
- Caches